lib/testsuite/index.js

```diff
@@ -159,7 +159,11 @@ class TestSuite {
     }
 
     if (this.validateNightwatchInspectorCriteria()) {
-      this.inspectorServer = new NightwatchInspectorServer(this.client);
+      if (this.globalsInstance.inspectorServer) {
+        this.globalsInstance.inspectorServer.closeSocket();
+      }
+
+      this.globalsInstance.inspectorServer = new NightwatchInspectorServer(this.client);
```

Comment:
It is up to you whether or not you want to follow this suggestion for refactoring.

```javascript
  if (this.validateNightwatchInspectorCriteria()) {
    this.closeInspectorServerSocket();
    this.createNightwatchInspectorServer();
  }
  
closeInspectorServerSocket() {
  this.globalsInstance.inspectorServer?.closeSocket();
}

createNightwatchInspectorServer() {
  this.globalsInstance.inspectorServer = new NightwatchInspectorServer(this.client);
}
```

In my opinion, this version appears much cleaner and easier to read. The `closeInspectorServerSocket()` method is responsible for closing the socket, while the `createNightwatchInspectorServer()` method is responsible for creating the inspector server.

Reply:
calling functions is overkill in my opinion, but i like the idea of optional chaining. Updated